examples/simple.js

```diff
@@ -1,17 +1,224 @@
+          if (child.hasOwnProperty('children')){
+            child.children.forEach(function(grand_child){
+              var cat_info = grand_child.path + ',' + grand_child.id+'\n';
+              fs.appendFileSync('subcategory_info.txt', cat_info);
+            })
+          }
+          else {
+            var cat_info = child.path + ',' + child.id+'\n';
+            fs.appendFileSync('subcategory_info.txt', cat_info);
+          }
+        })
+      }
+      else {
+        var cat_info = category.name + ' ' + category.id+'\n';
+        fs.appendFileSync('subcategory_info.txt', cat_info);
+      }
+    })
+  });
+}
+
+
+function writeCategoryIdsToFile(){
+  /*
+    writes category id and names to file
+  */
+  walmart.taxonomy().then(function(data){
+    data.categories.forEach(function(category){
+      var cat_info = category.name + ',' + category.id+'\n';
+
+      fs.appendFileSync('category_info.txt', cat_info);
+    })
+  });
+}
+
+function writeCategoryIdsOnlyToFile(){
+  /*
+    writes category ids that can be used to get clearance information
+    should be called once in a given period since category_ids rarely change daily
+  */
+  walmart.taxonomy().then(function(data){
+    data.categories.forEach(function(category){
+      var cat_info = category.id+'\n';
+      fs.appendFileSync('category_ids.txt', cat_info);
+    })
+  });
+}
+
+function extractProductInfo(product){
+  var twoThreeDayShippingRate = Object.is(product.twoThreeDayShippingRate, undefined) ? '-1.00' : product.twoThreeDayShippingRate; 
+  var msrp = Object.is(product.msrp, undefined) ? '-1.00' : product.msrp;
+  var product_info=product.itemId+','+product.upc+','+product.name+','+product.brandName+','
+    +msrp+','+product.salePrice+','+product.standardShipRate+','
+    +twoThreeDayShippingRate+','+product.availableOnline+','+product.categoryNode
+    +','+product.stock+','+product.freeShippingOver50Dollars+'\n';
+  return product_info;
+}
+
+
+function getRecursiveResponse(response){
+  var results_array = [];
+  if (response != undefined && response.hasOwnProperty('items')){
+    response.items.forEach(function(item){
+      // uses upc for identification
+      if (item.hasOwnProperty('upc')){
+        results_array.push(extractProductInfo(item));
+      }
+    })
+    if (response.nextPage) {
+      var new_page_response = walmart.getNewPage(response.nextPage);
+      new_page_response.then(function(resp){
+        results_array = results_array.concat(results_array, getRecursiveResponse(resp));
+      })      
+    }
+    results_array.forEach(function(item){
+      fs.appendFile('special_feeds_items.txt', item, (err) => {
+        if (err) throw err;
+      });
+    })
+    
+    return;
+  }
+}
+
+
+/*
+function that serially calls Promise functions and waits for each of them to be resolved
+Reference: https://hackernoon.com/functional-javascript-resolving-promises-sequentially-7aac18c4431e
+*/
+const promiseSerial = cat_array =>
```

Comment:
I didn't realize node supported ES6 out of the box. In general, it's a good idea to declare all variables that won't be changing as `const` and all mutable variables as `let`. This is the only place you use a const. I'm not gonna make you go and change it everywhere, but wanted to point it out so we can declare variables correctly in the future.